### PR TITLE
Feature/custom ldap attr

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Allows the definition of custom LDAP attributes directly in the configuration for use in the authentication rules
# Impacts

Reduces custom code
# Dependencies

feature/merged-list (#664)
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Added the ability to define custom LDAP attributes in the configuration

### CODE REVIEW

**Comment by jrouzierinverse** [comment]
Update the translation file

File: lib/pf/Authentication/Source/LDAPSource.pm [hunk 1]
```diff
@@ -53,8 +54,9 @@ sub available_attributes {
   my $self = shift;
 
   my $super_attributes = $self->SUPER::available_attributes;
-  my @ldap_attributes = map { { value => $_, type => $Conditions::LDAP_ATTRIBUTE } }
-    ("uid", "cn", "department", "displayName", "distinguishedName", "givenName", "memberOf", "sn", "eduPersonPrimaryAffiliation", "mail", "postOfficeBox", "description", "groupMembership");
+  my @custom_attributes = @{$Config{advanced}->{custom_ldap_attributes}};
+  my @attributes = (("uid", "cn", "department", "displayName", "distinguishedName", "givenName", "memberOf", "sn", "eduPersonPrimaryAffiliation", "mail", "postOfficeBox", "description", "groupMembership"), @custom_attributes);
```
Comment (cgx) [inline lib/pf/Authentication/Source/LDAPSource.pm]: You should extract unique attributes only (`List::MoreUtils::uniq`).

Comment (julsemaan) [inline lib/pf/Authentication/Source/LDAPSource.pm]: Done

Comment (jrouzierinverse) [inline lib/pf/Authentication/Source/LDAPSource.pm]: Move the standard list into a constant

File: lib/pf/Authentication/constants.pm [hunk 2]
```diff
@@ -16,6 +16,8 @@ use warnings;
 
 use Readonly;
 
+our @LDAP_ATTRIBUTES = ("uid", "cn", "department", "displayName", "distinguishedName", "givenName", "memberOf", "sn", "eduPersonPrimaryAffiliation", "mail", "postOfficeBox", "description", "groupMembership");
```
Comment (dwlfrth) [inline lib/pf/Authentication/constants.pm]: What is my purpose ?

**Comment by dwlfrth** [comment]
Also, please rebase so that we can benefit of the merged-list feature.

**Pull request merged**